lib/applications/resolvers.ts

```diff
@@ -547,3 +549,95 @@ export const createReplacementApplication: Resolver = async (_, args, { prisma }
     ok: true,
   };
 };
+
+export const generateApplicantsReport: Resolver = async (_, args, { prisma }) => {
+  const {
+    input: { startDate, endDate, columns },
+  } = args;
+
+  // NOTETOSELF: Consider making columns an object instead to avoid array look up inefficiency + clean up code
+
+  const applications: GenerateApplicantsReportApplications[] = await prisma.application.findMany({
+    where: {
+      createdAt: {
+        gte: new Date(startDate),
+        // TODO; Figure out why lte isn't working
+        // lte: new Date(endDate),
+      },
+    },
+    select: {
+      // TODO?: Replace with function/object in constants
+      rcdUserId: columns.includes(ApplicationsReportColumn.UserId),
```

Comment:
I think making columns an object instead of an array could be better. We'll avoid inefficient array lookup and it would make the code cleaner:
e.g
```ts
 select: {
      rcdUserId: columns.userId,
```

Reply:
Converted the array in the resolver based on oustan's suggestion